org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/PresetEditingWizardEventPage.java

```diff
@@ -67,20 +80,119 @@ public void createControl(Composite parent) {
 		Composite container = new Composite(sc, SWT.NONE);
 		sc.setContent(container);
 
-		// TODO: create event page control here
 		container.setLayout(new FillLayout());
-		new Label(container, SWT.NONE).setText("TODO: create event page control here");
+
+		createEventTableContainer(container);
 
 		sc.setExpandHorizontal(true);
 		sc.setExpandVertical(true);
 		sc.setMinSize(container.computeSize(SWT.DEFAULT, SWT.DEFAULT));
 		setControl(sc);
+
+		populateUi();
+	}
+
+	private Composite createEventTableContainer(Composite parent) {
+		Composite container = new Composite(parent, SWT.NONE);
+		container.setLayout(new FillLayout());
+
+		tableInspector = new TableInspector(container,
+				TableInspector.MULTI | TableInspector.SHOW_HEADER | TableInspector.ADD_BUTTON
+						| TableInspector.EDIT_BUTTON | TableInspector.DUPLICATE_BUTTON | TableInspector.REMOVE_BUTTON) {
+			@Override
+			protected void addColumns() {
+				addColumn(LABEL_ID_COLUMN, ID_ID_COLUMN, new EventTableLabelProvider() {
+					@Override
+					protected String doGetText(IEvent event) {
+						return event.getId();
+					}
+				});
+
+				addColumn(LABEL_NAME_COLUMN, ID_NAME_COLUMN, new EventTableLabelProvider() {
+					@Override
+					protected String doGetText(IEvent event) {
+						return event.getName();
+					}
+				});
+			}
+
+			@Override
+			protected void onAddButtonSelected(IStructuredSelection selection) {
+				openEventEditingWizardFor(new Event());
+				tableInspector.getViewer().refresh();
+			}
+
+			@Override
+			protected void onEditButtonSelected(IStructuredSelection selection) {
+				openEventEditingWizardFor((IEvent) selection.getFirstElement());
+				tableInspector.getViewer().refresh();
+			}
+
+			@Override
+			protected void onDuplicateButtonSelected(IStructuredSelection selection) {
+				// TODO: create a copy properly
+				IEvent original = (IEvent) selection.getFirstElement();
+				IEvent duplicate = new Event();
+				duplicate.setId(original.getId() + ".copy");
+				duplicate.setName("Copy of " + original.getName());
+				preset.addEvent(duplicate);
+				tableInspector.getViewer().refresh();
+			}
+
+			@Override
+			protected void onRemoveButtonSelected(IStructuredSelection selection) {
+				preset.removeEvent((IEvent) selection.getFirstElement());
+				tableInspector.getViewer().refresh();
+			}
+		};
+		tableInspector.setContentProvider(new EventTableContentProvider());
+
+		return container;
+	}
+
+	private void populateUi() {
+		tableInspector.setInput(preset);
 	}
 
-	// TODO: call this function when "New" or "Edit" button clicked
 	private void openEventEditingWizardFor(IEvent event) {
-		if (DialogToolkit.openWizardWithHelp(new EventEditingWizard(event))) {
-			// TODO: save the modified event to the preset
+		while (DialogToolkit.openWizardWithHelp(new EventEditingWizard(event))) {
+			try {
+				// TODO: save the modified event to the preset
+				preset.addEvent(event);
```

Comment:
on edit this fails because it tries to add the event again, when it should only be being modified.  You get stuck in the error message "An event with the same id already exists".